Collaborator-Guide.md

```diff
@@ -18,21 +18,90 @@ current release stream.
 Use the correspondingly named branch, e.g. `5.0`, for anything intended for
 a future release of Express.
 
-## Steps for contributing
+## Submitting a Pull Request (PR)
 
-1. [Create an issue](https://github.com/expressjs/express/issues/new) for the
+Before you submit your Pull Request (PR) consider the following guidelines:
+
+1. Search [GitHub](https://github.com/expressjs/express/pulls) for an open or closed PR that relates to your submission. You don't want to duplicate effort.
+
+2. [Create an issue](https://github.com/expressjs/express/issues/new) for the
```

Comment:
If the user is about to submit a PR, it would be best to not open an issue then open a PR. The issue this creates is that some folks may start working on the PR, but the opener was already working on it. I suggest a user either open a PR or an issue, but not both.